components/empty-state/test/empty-state-text.axe.js

```diff
@@ -0,0 +1,23 @@
+import '../empty-state-text.js';
+import '../../link/link.js';
+import { expect, fixture, html } from '@open-wc/testing';
+
+describe ('d2l-empty-state-text', () => {
+
+	it('normal', async() => {
+		const el = await fixture(html`
+            <d2l-empty-state-text description='There are currently no courses here.' action-text='Create a Course'></d2l-empty-state-text>
+        `);
+		await expect(el).to.be.accessible();
+	});
+
+	it('custom link', async() => {
+		const el = await fixture(html`
+            <d2l-empty-state-text description='There are currently no courses here.'>
+				<d2l-link  href="https://www.d2l.com/">Create a Course</d2l-link>
```

Comment:
I wasn't in the planning sessions for this, but I'm curious about how we landed on using a slot for the link use-case? It seems like for the majority of uses of this, the "Create a X" action will perform a navigation, and therefore this should almost always be a link (`<a>` tag) vs. performing an action (like "Clear Search" in the design) where it should be a button.

I think I'm just making the argument that the link-ness or button-ness should just be built into the component's API first-class using attributes, instead of relying on a slot where they can throw anything in there.

A proposal -- this would render a link:
```html
<d2l-empty-state-text description="blah" action-text="Create" action-href="/create/thing">
```

And this would render a button:
```html
<d2l-empty-state-text description="blah" action-text="Create" @empty-state-action="${this.handleClick}">
```

Reply:
Admittedly, it was not one of the things we expanded upon during the planning session. I was actually thinking similarly about this and was going to propose that we change the default action in the component to be a link rather than a button but I really like your suggestion to alter the API and remove the need to use a slot at all.

There was some discussion on whether we would want to support other actions like a primary button on a page with no other actions, but I think that would be more for the illustration empty-state component.

I will bring this idea to the team. Thanks!

Reply:
Cool, sounds good! There are other options as well... like you could have separate components `<d2l-empty-state-text-link>`, `<d2l-empty-state-text-button>`, or could leverage a `<d2l-empty-state-text action-type="link|button|link-primary">`. The `action-type` approach can get weird because they can choose `link` without passing an `href` or `button` AND pass an `href`, so we typically try to avoid that kind of API.

Reply:
The reason the slot was suggested is that the subtle button is the primary use-case we expect, but we would like the ability to override, sometimes with a primary button, but possibly even with a button-menu. The slot would enable them to do that without surfacing those semantics on the empty-state component itself. Honestly, I feel conflicted about even baking the subtle button in, but Jeff & Neroli really want consumers to follow that pattern.  If we go the slot route, we can target and hide slotted elements that we don't want them to put in there. 

If we go with different components, we can lock things down better, but we will have the two variants (due to very different layouts) multiplied by the different types of actions. 🤷 Lots of components.

Completely agree on overloading a single component - it can get a little messy.

Anyway, that was the original thinking, and we can certainly pivot from here.

Reply:
> ...the subtle button is the primary use-case we expect

I guess I'm questioning that? Only based on the examples in the design though, so if there's better data out there let's use that. But from the design, "Create Learning Paths", "View All Activities", "Create an Assignment" should be all be links because they navigate and only "Clear Search" should be a button since it's an action. I think this is why designers keep asking for a `<d2l-link-subtle>`! :)